File: Samples/Draw and print PDF/PrintPdf/VB.NET/WinForms/MainForm.vb
Imports System.Windows.Forms
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Partial Public Class MainForm
        Inherits Form

        Private Const FitPageIndex = 0

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Function GetPrintSize() As PrintSize
            Return If(printSize.SelectedIndex = FitPageIndex, Samples.PrintSize.FitPage, Samples.PrintSize.ActualSize)
        End Function

        Private Sub PrintButton_Click(sender As Object, e As EventArgs) Handles printButton.Click
            ProcessExistingPdfDocument(New Func(Of PdfDocument, PrintSize, DialogResult)(AddressOf ShowPrintDialog))
        End Sub

        Private Sub PreviewButton_Click(sender As Object, e As EventArgs) Handles previewButton.Click
            ProcessExistingPdfDocument(New Func(Of PdfDocument, PrintSize, DialogResult)(AddressOf ShowPrintPreview))
        End Sub

        Private Sub ProcessExistingPdfDocument(action As Func(Of PdfDocument, PrintSize, DialogResult))
            Using dlg As New OpenFileDialog()
                dlg.Filter = "PDF files (*.pdf)|*.pdf"

                If dlg.ShowDialog() = DialogResult.OK Then
                    ' NOTE:
                    ' Without a license, the library won't allow you to create or read PDF documents.
                    ' To get a free time-limited license key, use the form on
                    ' https://bitmiracle.com/pdf-library/download

                    LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

                    Using pdf As PdfDocument = New PdfDocument(dlg.FileName)
                        action(pdf, GetPrintSize())
                    End Using
                End If
            End Using
        End Sub
    End Class
End Namespace

File: Samples/Draw and print PDF/PrintPdf/C#/WinForms/MainForm.cs
using System;
using System.Windows.Forms;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public partial class MainForm : Form
    {
        private const int FitPageIndex = 0;

        public MainForm()
        {
            InitializeComponent();
        }

        private PrintSize GetPrintSize()
        {
            return (printSize.SelectedIndex == FitPageIndex) ? PrintSize.FitPage : PrintSize.ActualSize;
        }

        private void PrintButton_Click(object sender, EventArgs e)
        {
            ProcessExistingPdfDocument(PdfPrintHelper.ShowPrintDialog);
        }

        private void PreviewButton_Click(object sender, EventArgs e)
        {
            ProcessExistingPdfDocument(PdfPrintHelper.ShowPrintPreview);
        }

        private void ProcessExistingPdfDocument(Func<PdfDocument, PrintSize, DialogResult> action)
        {
            using var dlg = new OpenFileDialog();
            dlg.Filter = "PDF files (*.pdf)|*.pdf";

            if (dlg.ShowDialog() == DialogResult.OK)
            {
                // NOTE:
                // Without a license, the library won't allow you to create or read PDF documents.
                // To get a free time-limited license key, use the form on
                // https://bitmiracle.com/pdf-library/download

                LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

                using var pdf = new PdfDocument(dlg.FileName);
                action(pdf, GetPrintSize());
            }
        }
    }
}
